frontend/public/components/RBAC/bindings.jsx

```diff
@@ -295,28 +299,58 @@ ListDropdown.propTypes = {
   selectedKey: PropTypes.string,
   selectedKeyKind: PropTypes.string,
   fixed: PropTypes.bool,
-  kinds: PropTypes.arrayOf(PropTypes.string).isRequired,
+  resources: PropTypes.arrayOf(PropTypes.shape({
+    kind: PropTypes.string.isRequired,
+    namespace: PropTypes.string,
+  })).isRequired,
   placeholder: PropTypes.string,
 };
 
-const NsDropdown = props => <ListDropdown {...props} desc="Namespaces" kinds={['Namespace']} selectedKeyKind="Namespace" placeholder="Select namespace" />;
+const NsDropdown_ = props => {
+  const openshiftFlag = props.flags[FLAGS.OPENSHIFT];
+  if (flagPending(openshiftFlag)) {
+    return null;
+  }
+  const kind = openshiftFlag ? 'Project' : 'Namespace';
+  const resources = [{ kind }];
+  return <ListDropdown {...props} desc="Namespaces" resources={resources} selectedKeyKind={kind} placeholder="Select namespace" />;
+};
+const NsDropdown = connectToFlags(FLAGS.OPENSHIFT)(NsDropdown_);
+
+const NsRoleDropdown_ = props => {
+  const openshiftFlag = props.flags[FLAGS.OPENSHIFT];
+  if (flagPending(openshiftFlag)) {
+    return null;
+  }
+
+  const roleFilter = role => !isSystemRole(role);
+
+  let kinds;
```

Comment:
only nit, I bet we could `const` this as well. :)

Reply:
knew you were going to comment on that :)

i don't know a good way to do it without nested ternaries, which i want to avoid

Reply:
Only offering for the giggles of it:

```JavaScript 
const kinds = props.fixed 
  ? [props.selectedKeyKind] 
  : props.namespace 
    ? ['Role', 'ClusterRole']
    : ['ClusterRole'];
```
surely everyone loves nested ternary?  😜

Reply:
lol the timing on that was awesome.